packages/venia-concept/src/components/ToastContainer/toast.js

```diff
@@ -0,0 +1,54 @@
+import React, { useCallback } from 'react';
+import { func } from 'prop-types';
+import defaultClasses from './toast.css';
+import { useToastActions } from '@magento/peregrine/src/Toasts/useToastActions';
+import { mergeClasses } from 'src/classify';
+import Icon from 'src/components/Icon';
+
+const Toast = props => {
+    const { id, type, icon, message, actionText, dismissable } = props;
+    const classes = mergeClasses(defaultClasses, {});
+    const { removeToast } = useToastActions();
+
+    // TODO: Can these callbacks be defined within the addToast hook?
```

Comment:
If I can do this it won't be necessary for a toast to know its own `id`.

Reply:
But if we do move this to the toast action it becomes that much more opinionated about what happens when you click on an action or dismiss the toast.

Maybe there is a compromise that dismiss handler is made in the action but the action handler is defined within the toast, as that belongs to the consumer (in this case, Venia).